crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+        let result_start = buffer.ptr;
+
+        match selector {
+            "call" => {
+                let contract_address = inputs[0].to_field_element();
+                let function_name = as_cairo_short_string(&inputs[1])
+                    .expect("Failed to convert function name to short string");
+                let calldata_length = inputs[2]
+                    .to_usize()
+                    .expect("Failed to convert calldata length to usize");
+                let calldata = Vec::from(&inputs[3..(3 + calldata_length)]);
+                let calldata_felts: Vec<FieldElement> = calldata
+                    .iter()
+                    .map(StarknetConversions::to_field_element)
+                    .collect();
+
+                let call_response = self.runtime.block_on(call::call(
+                    contract_address,
+                    &function_name,
+                    calldata_felts,
+                    self.provider,
+                    &BlockId::Tag(Pending),
+                ))?;
+
+                buffer
+                    .write(call_response.data.len())
+                    .expect("Failed to insert data length");
+
+                buffer
+                    .write_data(
+                        call_response
+                            .data
+                            .iter()
+                            .map(StarknetConversions::to_felt252),
+                    )
+                    .expect("Failed to insert data");
+
+                Ok(())
+            }
+            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+        }?;
+
+        let result_end = buffer.ptr;
+        insert_value_to_cellref!(vm, output_start, result_start)?;
+        insert_value_to_cellref!(vm, output_end, result_end)?;
+
+        Ok(())
+    }
+}
+
+pub fn run(
+    script_path: &Utf8PathBuf,
+    provider: &JsonRpcClient<HttpTransport>,
+    runtime: Runtime,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
+) -> Result<ScriptResponse> {
+    check_compiler_path(true, Path::new(&script_path))?;
+
+    let metadata = parse_scarb_metadata(path_to_scarb_toml)?;
+
+    let corelib = corelib_for_package(&metadata, &metadata.workspace.members[0])?;
+    let db = &mut RootDatabase::builder().build()?;
+
+    init_dev_corelib(db, PathBuf::from(corelib));
+
+    let main_crate_ids = setup_project(db, Path::new(&script_path))?;
+    if DiagnosticsReporter::stderr().check(db) {
+        anyhow::bail!("failed to compile: {}", script_path);
+    }
+
+    let sierra_program = db
+        .get_sierra_program(main_crate_ids.clone())
+        .to_option()
+        .with_context(|| "Compilation failed without any diagnostics.")?;
+    let replacer = DebugReplacer { db };
+
+    let runner = SierraCasmRunner::new(
+        replacer.apply(&sierra_program),
+        Some(MetadataComputationConfig::default()),
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
```

Comment:
Why is it needed

Reply:
You mean why a replacer is needed?

Reply:
not relevant anymore, since the compilation is now performed by scarb